### PR TITLE
Updated execute_with_gradients to support multiple variables and structures with gradient indexing

### PR DESCRIPTION
**Pull request by VedPatwardhan**

### CODE REVIEW

File: ivy/functional/backends/jax/gradients.py [hunk 1]
```diff
@@ -31,28 +36,28 @@ def variable_data(x):
...
+        grads = grad_fn(xs)
     else:
-        grads = grad_func(xs)
-        grads = ivy.to_ivy(grads)
-    if not retain_grads:
-        y = ivy.stop_gradient(y)
-    return (y, grads, *rest)
+        grad_fn = jax.jacrev(lambda x: _forward_fn(x, func))
+        grads_ = grad_fn(xs)
+        for i, grad in enumerate(grads_):
```
Comment (djl11) [inline ivy/functional/backends/jax/gradients.py:55]: can be done in one line:
`grads = {arr_idxs[i]: grad for i, grad in enumerate(grads_)}`

File: ivy/functional/backends/jax/gradients.py [hunk 2]
```diff
@@ -31,28 +36,28 @@ def variable_data(x):
...
-        rest = tuple()
-        grad_fn = lambda x_in: ivy.to_native(ivy.reshape(func(ivy.to_ivy(x_in)), []))
-    grad_func = jax.grad(grad_fn)
     xs = ivy.to_native(xs)
-    if isinstance(xs, ivy.Container):
-        grads = grad_func(xs)
-        grads = ivy.to_ivy(grads)
-        grads = Container(grads)
+    arr_idxs, arr_values = _get_native_arrays_and_indices(func_ret)
+    grads = {}
```
Comment (djl11) [inline ivy/functional/backends/jax/gradients.py:43]: not needed with one-liner below

**Review by djl11 — CHANGES_REQUESTED** [review]
small changes requested (only for `jax` but I think similar changes can be made for most backends), then you can go ahead and merge, provided that quite you're confident it's been implemented correctly. Thanks!
